study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,190 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
 - Purpose-built well for pure HTML.
 - There are many granular controls, concerning things like parent elements and sizing and alignment inheritance.
 - The component comes with events, for when the toast opens and the alignment changes (*not for when it closes though*).
 - Only one `<paper-toast>` will be visible on the screen at a time. If you trigger a second it will replace the first.
 - **Warning**: `<paper-toast>` is affected by the stacking context of its container, so to guarantee it shows up on the top layer it must be placed at the top level (`<body>`) element.
     - How to properly handle top layer is an ongoing debate (see [here](https://github.com/whatwg/html/issues/897#issuecomment-198512778))
 
-## Material-UI Snackbar
+## [Material-UI Snackbar](https://material-ui.com/api/snackbar/)
 
 *TODO: Fill in study for this library*
 
-## Android Toast
+## [Android `Toast`](https://developer.android.com/guide/topics/ui/notifiers/toasts)
+Android `Toast` is a simple toast implementation built by the Android team for usage in Android apps.
 
-*TODO: Fill in study for this library*
+### Design & Sample Code
+**Note**: Since Android development is done in Kotlin all code samples will be Kotlin.
 
-## react-toastify
+To trigger displaying a simple Android `Toast`, put this in your Kotlin:
 
-*TODO: Fill in study for this library*
+```kotlin
+val text = "Hello toast!"
+val duration = Toast.LENGTH_SHORT
+
+val toast = Toast.makeText(applicationContext, text, duration)
+toast.show()
+```
+
+This creates a toast object filled with a context, some text, and a duration, which is triggered to display by the `toast.show()` method.
+This will show up on the bottom of the screen by default.
+To control the position of the toast, set the gravity, like this:
+
+```kotlin
+toast.setGravity(Gravity.TOP or Gravity.LEFT, 0, 0)
+```
+
+This sample will move the toast to the top left of the screen.
+The final two parameters are used to offset the toast from its center of gravity, so this:
+
+```kotlin
+toast.setGravity(Gravity.TOP or Gravity.LEFT, 10, 10)
+```
+
+Will show the toast further down and further right.
+The toast can also be injected with a custom layout, which requires more complex Kotlin, explained [here](https://developer.android.com/guide/topics/ui/notifiers/toasts#CustomToastView).
+The `text`, `view`, `gravity`, and `duration` of the toast are all the elements the developer can control when working with this very straightforward component.
+
+
+### Notable Features & Details
+
+#### Gravity
+The library takes the Android approach to positioning,
+by letting the user define the Gravity of the element,
+along with offset values for the axes.
+
+#### Duration Constants
+The library comes with baked in constants for displaying the toast for long and short periods of time.
+
+### Takeaways
+- High opinion, low control.
+- The API for the `Toast` object provides getters and setters for each of `text`, `view`, `gravity`, `offset` and `duration`.
```

Comment:
Does this mean they can be changed dynamically, while the toast is displaying? I hadn't thought of that before, but I guess it makes sense. I'm not sure if it's worth mentioning explicitly, but I found it interesting.

Reply:
To be honest, I'm not sure. The way the toast itself works is that you create and configure the object, then show it. 

What I think is probably more likely is that `show()` just shows the toast with the previously configured options, and setting the options while it displays will only affect the toast the next time you call `show()`.